src/pages/main/tasks/task/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { ITask } from '../../../../api/requests';
+import { reduxProject, StateReduxInterface } from '../../../../state';
+import { useDispatch, useSelector } from 'react-redux';
+import { IBoard } from '../../boards/board';
+
+export const MainCard = function (task: ITask) {
+  const data = useSelector((state: StateReduxInterface) => state);
+  const dispatch = useDispatch();
+  const router = useNavigate();
+  const { t } = useTranslation();
+  const boardLink = () => {
+    dispatch(
+      reduxProject({
+        board: data.boards.find((e) => e._id === task.boardId) as IBoard,
+        tasks: [],
+        columns: [],
+      })
+    );
```

Comment:
this action sounds more like `dispatch(setCurrentBoard(task.boardId))`, not `dispatch(reduxProject())`

avoid passing all values of your store at once, instead just pass the one that should be updated
in this case, you could just pass a `boardId` and handle the rest in your store logic